### PR TITLE
Added override/ & tools/ to PHPStan Check

### LINKED ISSUE #16471: Use static analyser to improve code quality
Let's integrate [PHPStan](https://github.com/phpstan/phpstan) or [Psalm](https://github.com/vimeo/psalm) into our CI to improve code quality.

[EDIT by @Progi1984]
:notebook: Applied
- [x] on `admin-dev/`
  - #24678
- [x]  on `classes/` (445)
  - #25533
  - #25916
  - #26164
  - #26478
  - #26727
  - #26913
- [x]  on `config/` (25)
  - #26479
- [x]  on `controllers/` (799)
  - #26518
  - #26770
  - #26864
  - #27004
  - #27102
  - #27417
- [x]  on `install-dev/` (497)
  - #26819
- [x]  on `js/` (2)
  - #26744
- [x] on `override/`
  - #24466
- [x] on `src/` (above)
  - #19898
  - #20118 
  - #21241 
  - #21515 
  - #22547 
- [x] on `tools/`
  - #24466
- [x]  on `tests/` (312)
  - #26592
  - #27107
  - #27125
  - #27141
  - #27287
- [x]  on `tests-legacy/`
  - #27247
- [x] on `webservice/`
  - #23960

Final 
- [x] Update to latest version of PhpStan : #27723
- [x] Use `dynamicConstantNames` (https://phpstan.org/config-reference#constants) and remove `@phpstan-ignore-next-line` : #28060

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Added override/ & tools/ to PHPStan Check
| Type?             | improvement
| Category?         | PM
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #16471
| How to test?      | Github Actions are :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24466)

### CODE REVIEW

File: tools/build/Library/InstallUnpacker/classes/InstallManager.php [hunk 1]
```diff
@@ -185,9 +185,10 @@ private function replaceInstallFile($fileName)
      */
     private function deleteDirectoryWithItsContent($directoryPath)
     {
-        $deleteDirectoryContentResult = array_map(
+        $deleteDirectoryies = glob($directoryPath . DIRECTORY_SEPARATOR . '*.*');
```
Comment (matks) [inline tools/build/Library/InstallUnpacker/classes/InstallManager.php]:
```suggestion
        $directoriesToDelete = glob($directoryPath . DIRECTORY_SEPARATOR . '*.*');
```

File: tools/build/Library/InstallUnpacker/classes/InstallManager.php [hunk 2]
```diff
@@ -185,9 +185,10 @@ private function replaceInstallFile($fileName)
...
+        $deleteDirectoryContentResult = !$deleteDirectoryies ? false : array_map(
```
Comment (matks) [inline tools/build/Library/InstallUnpacker/classes/InstallManager.php]:
```suggestion
        $deleteDirectoryContentResult = !$directoriesToDelete ? false : array_map(
```

File: tools/build/Library/InstallUnpacker/classes/InstallManager.php [hunk 3]
```diff
@@ -185,9 +185,10 @@ private function replaceInstallFile($fileName)
...
             'unlink',
-            glob($directoryPath . DIRECTORY_SEPARATOR . '*.*')
+            $deleteDirectoryies
```
Comment (matks) [inline tools/build/Library/InstallUnpacker/classes/InstallManager.php]:
```suggestion
            $directoriesToDelete
```

File: tools/foreignkeyGenerator/changes.php [hunk 4]
```diff
@@ -122,12 +122,6 @@
             '#unsigned' => true,
         ),
     ),
-    'tax_rules_group' => array(
```
Comment (matks) [inline tools/foreignkeyGenerator/changes.php:125]: Is this related to PHPStan 🤔  ? If yes can you explain how this is marked as false by a static analysis

File: tools/build/CreateRelease.php [hunk 5]
```diff
@@ -88,18 +88,6 @@
     exit(0);
 }
 
-foreach ($releaseOptions as $optionName => $option) {
```
Comment (matks) [inline tools/build/CreateRelease.php:91]: Hello, so PHPStan says this is dead code? `$option['required']` and `$userOptions[$optionName]` cannot be set right?

**Review by matks — COMMENTED** [review]
A few questions to make sure we break nothing 😉  also profiling will have to be tested by someone

File: tools/foreignkeyGenerator/changes.php (hunk 4 above)
Comment (Progi1984) [reply tools/foreignkeyGenerator/changes.php:125]: This key is a duplicated to the line 27.

File: tools/build/CreateRelease.php (hunk 5 above)
Comment (Progi1984) [reply tools/build/CreateRelease.php:91]: Affirmative. Why ? Because `$required` will be always defined to `false` so it never go in `if` section.

File: phpstan.neon.dist [hunk 6]
```diff
@@ -14,6 +16,11 @@ parameters:
 		- '#^Call to an undefined static method Attribute\:\:checkAttributeQty\(\)\.$#'
 		- '#^Call to an undefined static method Attribute\:\:getAttributeMinimalQty\(\)\.$#'
 		- '#^Call to an undefined static method Attribute\:\:getAttributes\(\)\.$#'
+		## Relative to Profiler
+		- '#^Access to an undefined property Db\:\:\$queries\.#'
```
Comment (PierreRambaud) [inline phpstan.neon.dist]: Maybe fixing properties instead?

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @Progi1984
